PROTOCOL.md

```diff
@@ -0,0 +1,74 @@
+# GraphQL over WebSocket Protocol
+
+## Client-server communication
+
+Each message has a `type` field, which defined in the protocol of this package, as well as associated fields inside `payload` field, depending on the message type.
+
+### Client -> Server
+
+#### GQL_CONNECTION_INIT
+Client sends this message after connecting, this triggers `onConnect` on the server.
+- `payload: Object` : optional parameters that the client specifies in `connectionParams`
+
+#### GQL_CONNECTION_TERMINATE
+Client sends this message to terminate the connection.
+
+#### GQL_START
+Client sends this message to execute GraphQL operation
```

Comment:
This should be:
- id: string: The id of the GraphQL operation to start
- payload: Object
  -- `query: string` : GraphQL operation as string or parsed GraphQL document node
  -- `variables?: Object` : Object with GraphQL variables
  -- `operationName?: string` : GraphQL operation name
  -- `context?: any` : Execution context for the operation